compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -1519,8 +1519,8 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
 
             // FIXME(or_patterns; matthewjasper) Try to be more aggressive here.
             can_merge &= subcandidate.subcandidates.is_empty()
-                && subcandidate.bindings.is_empty()
-                && subcandidate.ascriptions.is_empty();
+                && subcandidate.extra_data.bindings.is_empty()
+                && subcandidate.extra_data.ascriptions.is_empty();
```

Comment:
Will the future information you add here be similar to these two lists and checked here? If so, maybe a single `is_empty` method on the struct could be useful

Reply:
Good shout :+1: